### PR TITLE
[Bug](array_product) Fix array_product for ARRAY<DECIMAL>

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

img_v2_01a0366b-20ed-444d-8e42-fd6cc96df24g

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.56 seconds
 stream load tsv:          475 seconds loaded 74807831229 Bytes, about 150 MB/s
 stream load json:         23 seconds loaded 2358488459 Bytes, about 97 MB/s
 stream load orc:          74 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230322124134_clickbench_pr_118676.html

File: be/src/vec/aggregate_functions/aggregate_function_product.h [hunk 1]
```diff
@@ -77,6 +77,30 @@ struct AggregateFunctionProductData<Decimal128> {
...
 };
 
+template <>
+struct AggregateFunctionProductData<Decimal128I> {
+    Decimal128I product {};
+
+    template <typename NestedType>
+    void add(Decimal<NestedType> value, UInt32 scale) {
+        product *= value;
+        product /= DataTypeDecimal<Decimal128I>::get_scale_multiplier(scale);
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_product.h:87]: why directly pass `DataTypeDecimal<Decimal128I>::get_scale_multiplier(scale)` to reduce the function call ？

**Comment by Gabriel39** [comment]
run buildall

**Review by HappenLee — DISMISSED** [review]
LGTM

**Comment by Gabriel39** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**